omnibus/files/private-chef-cookbooks/private-chef/templates/default/nginx/scripts/check_token_using_endpoint.lua.erb

```diff
@@ -0,0 +1,24 @@
+ngx.req.read_body()
+local res
+
+if ngx.var.request_method == "POST" then
+  meth = ngx.HTTP_POST
+else
+  meth = ngx.HTTP_GET
+end
+
+res = ngx.location.capture("/organizations/" .. ngx.var.request_org .. "/validate" .. ngx.var.request_uri,
+    { method = meth,
+      always_forward_body = true,
+      copy_all_vars = true,
+      vars = { request_org = ngx.var.request_org, upstream = "http://opscode_erchef" }
+    })
+
+if res.status == ngx.HTTP_OK then
+  ngx.var.upstream = '<%= URI.parse(node['private_chef']['data_collector']['root_url']).scheme %>://data-collector<%= URI.parse(node['private_chef']['data_collector']['root_url']).path %>'
+  ngx.req.set_header("x-data-collector-token", "<%= node['private_chef']['data_collector']['token'] %>")
+  ngx.req.set_header("x-data-collector-auth", "version=1.0")
+  return
+end
```

Comment:
We will get the `res.status` as the nginx response @stevendanna but we will not have extra details logged. However we should be able to find more info from the erchef logs. 

I feel pretty good with the current level but it's possible to add an additional log with the body or the response from `/validate`.